datashader/spatial/viewshed.py

```diff
@@ -31,13 +31,14 @@
 
 # this value is returned by findMaxValueWithinDist() if there is no key within
 # that distance
-SMALLEST_GRAD = -9999999999999999999999.0
+SMALLEST_GRAD = float('-inf')
 
 PROJ_LL = 0
 PROJ_NONE = -1
 
 PI = math.pi
 
+#TODO: refactor NAN value...
 NAN = -9999999999999999
```

Comment:
Is there a good reason not to simply use NaN and test for it with isnan?  I don't see anything special about the NAN in the code there that would require a separate implementation like this.

Reply:
@jbednar Yes...all hold overs from C++ code.  I marked them for refactor and didn't hit it yet, but let's hold off on merging this and I'll refactor today.

Reply:
Ok, thanks.

Reply:
On second look, this NAN value is import as a token NAN value for integer dtype arrays.

Reply:
Damned integers with no NaN support!